subs/pantry/src/Pantry/Types.hs

```diff
@@ -1446,6 +1529,207 @@ rpmToPairs (RawPackageMetadata mname mversion mtree mcabal) = concat
   , maybe [] (\cabal -> ["cabal-file" .= cabal]) mcabal
   ]
 
+parseAndResolvePackageLocation :: Path Abs Dir -> Value -> Parser (IO (NonEmpty PackageLocation))
+parseAndResolvePackageLocation rootDir v = do
+  (Unresolved unresolvedPL) <- parsePackageLocation v
+  pure $ unresolvedPL (Just rootDir)
+
+combineUnresolved :: Unresolved a -> Unresolved b -> Unresolved (a,b)
+combineUnresolved a b = do
+  ua <- a
+  ub <- b
+  pure (ua, ub)
+
+parseSingleObject :: Value -> Parser (Unresolved (PackageLocation, RawPackageLocation))
+parseSingleObject value = withObject "LockFile" (\obj -> do
+                                                   original <- obj .: "original"
+                                                   complete <- obj .: "complete"
+                                                   orig <- parseRPL original
+                                                   comp <- parsePImmutable complete
+                                                   pure $ combineUnresolved comp orig
+                                                   ) value
+
+
+parseLockFile ::
+       Value -> Parser [Unresolved (PackageLocation, RawPackageLocation)]
+parseLockFile value = withObject "LockFile" (\obj -> do
+                                               vals :: Value <- obj .: "dependencies"
+                                               xs <- withArray "LockFileArray" (\vec -> sequence $ Vector.map parseSingleObject vec) vals
+                                               pure $ Vector.toList xs
+                                            ) value
+
+resolveLockFile :: Path Abs Dir -> Value -> Parser (IO [(PackageLocation, RawPackageLocation)])
+resolveLockFile rootDir v = do
+  val <- parseLockFile v
+  let val' = sequence val
+      val'' = resolvePaths (Just rootDir) val'
+  pure val''
+
+loadLockFile :: Path Abs File -> IO [(PackageLocation, RawPackageLocation)]
```

Comment:
it's not clear why we have a `loadLockFile` in `pantry` but the main logic is in stack, I think we need to have a better separation of concerns

Reply:
Moved to `Stack.Load` module. Thanks.